rust/serving/src/source.rs

```diff
@@ -122,10 +122,10 @@ impl ServingSourceActor {
             if messages.len() >= count || Instant::now() >= timeout_at {
                 break;
             }
-            let message = match self.messages.try_recv() {
-                Ok(msg) => msg,
-                Err(mpsc::error::TryRecvError::Empty) => break,
-                Err(mpsc::error::TryRecvError::Disconnected) => {
+            let next_msg = self.messages.recv();
+            let message = match tokio::time::timeout_at(timeout_at, next_msg).await {
+                Ok(Some(msg)) => msg,
+                Ok(None) => {
                     // If we have collected at-least one message, we return those messages.
                     // The error will happen on all the subsequent read attempts too.
                     if messages.is_empty() {
```

Comment:
Why do we need to send an error when we don't have any messages?

Reply:
If the code reaches here, all the `Sender`s are already dropped. Any subsequent read requests also will fail.